File: scala-3.6/template-renderer/src/test/scala/com/rbleggi/templaterenderer/renderers/HTMLRendererTest.scala
package com.rbleggi.templaterenderer.renderers

import org.scalatest.funsuite.AnyFunSuite

class HTMLRendererTest extends AnyFunSuite {
  test("should generate valid HTML output") {
    val output = new HTMLRenderer().render(Map("title" -> "Test Title", "content" -> "Test Content"))
    assert(new String(output) ==
      """
        |<html>
        |<head><title>Test Title</title></head>
        |<body>
        |<h1>Test Title</h1>
        |<p>Test Content</p>
        |</body>
        |</html>
        |""".stripMargin)
  }
}

File: kotlin/template-renderer/src/test/kotlin/com/rbleggi/templaterenderer/service/renderer/HtmlRendererTest.kt
package com.rbleggi.templaterenderer.service.renderer

import com.rbleggi.templaterenderer.model.Document
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class HtmlRendererTest {

    @Test
    fun `renders the title as h1 and content as p wrapped in html`() {
        val output = String(HtmlRenderer().render(Document("My Page", "Hello World")))
        assertEquals("<html><body>\n<h1>My Page</h1>\n<p>Hello World</p>\n</body></html>", output)
    }
}
